### PR TITLE
Hadoop Development Cluster

### PR DESCRIPTION
**Pull request by jack-altiscale**
A modern development environment for debugging, testing and patching components in the Hadoop ecosystem. The main objective of this PR is to give back to the community a more approachable Hadoop and provide our team with greater agility in working with these products.

Following the setup procedures, one command will bring up a kerberized Hadoop docker cluster with the following services:
- MIT Kerberos 5
- Altiscale HDFS 2.7.4
- Altiscale YARN 2.7.4
- Altiscale Hive 2.1.1
- Altiscale Spark (with Jupyter notebook)

https://altiscale.atlassian.net/browse/BLUE-59

### CODE REVIEW

File: dev-support/docker-compose/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## Requirements
+- Docker v17.12.0 or greater (with Docker VM configured to 4 CPU and 8GB RAM is recommended)
+- OSX 10.13.3 or Linux
+
+## Setup
+Update your hosts file so the web ui links work for the rm, nm and spark:
+```
+#/etc/hosts
+
+127.0.0.1       nodemanager1
```
Comment (zoran-altiscale) [inline dev-support/docker-compose/README.md:15]: nice to add namenode and for each service add a comment about which port the service is on.

File: dev-support/docker-compose/files/.ipynb_checkpoints/go-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,89 @@
+{
```
Comment (zoran-altiscale) [inline dev-support/docker-compose/files/.ipynb_checkpoints/go-checkpoint.ipynb:1]: do you need this?

File: dev-support/docker-compose/files/livy/LICENSE [hunk 3]
```diff
@@ -0,0 +1,353 @@
+
```
Comment (zoran-altiscale) [inline dev-support/docker-compose/files/livy/LICENSE]: I would actually keep livy stuff outside of this branch - keeping stuff that can fetch livy seems fine to me, but I don't think copy from the other repo is a right thing to do. Writing a script that can git clone from other places seems fine to me, but again, we'll ask Allen about this.

**Review by zoran-altiscale — COMMENTED** [review]
I would recommend removing files/livy from this PR. And  figure out how to get these files in order to get livy working in a different way (not forking them which is not the right approach.

File: dev-support/docker-compose/files/livy/LICENSE (hunk 3 above)
Comment (zoran-altiscale) [reply dev-support/docker-compose/files/livy/LICENSE]: Still working on the review... clicked too soon...

**Review by zoran-altiscale — COMMENTED** [review]
There are lots of stuff here - thanks so much for hard work Jack! So, I suggest we make a PR for 2.7.5, and start developing this in a branch. It'll take long time to detangle what should be in here, and what not...

**Review by zoran-altiscale — APPROVED** [review]
Looks good to me for a start. Let's iterate over this and try to integrate it to our unit-test/integration-testing pipeline when generating our RPM.

Great work Jack! Thanks so much. 

LGTM++

**Comment by jozsef-verticloud** [comment]
Can we demo this on e.g. one of the brown bags?

**Comment by jack-altiscale** [comment]
Sure @jozsef-verticloud , let me know when you would like me to present this

**Comment by zoran-altiscale** [comment]
we put it as extra work - so my suggestion is to present this either next
Tuesday during regular demos or during Architectural meeting this week.

**Pull request merged**